### ISSUE TITLE
Lane detection in my own Dataset!

### ISSUE THREAD

**Issue by jiaofang1222** [wvangansbeke/LaneDetection_End2End#19]
Hi,I am very emotional about the work you have done.I want to run the model with my own Dataset.What parameters do I need to generate the in .json file? Is it just the "poly_params" parameter used?I only generated this parameter and can run the code, but the final prediction result is not ideal. Do you have any suggestions?

**Comment by wvangansbeke** [comment]
Hi @jiaofang1222,
Yes, you only need `poly_params` in order to train this network. However, stability might be an issue when you don't change the hyperparameters for your own dataset. If you still encounter problems, you might try one of the techniques (such as pretraining or a  segmentation proxy task) that I mentioned in the _Discussion_ section of the README.  

Kind regards,
Wouter

**Comment by jiaofang1222** [comment]
Thank you for your recovery.I still have some questions.After I run experiments with my own data set.When I tested with the training set, I found that there was no over-fitting. Is it missing some steps?

**Comment by wvangansbeke** [comment]
Hi @jiaofang1222,

Why do you think it would overfit? Overfitting happens when the validation loss is starting to increase instead of decreasing.

**Comment by jiaofang1222** [comment]
Sorry, there is a problem with my statement.Is it only used of "lines" and "poly_params" in the entire regression process and model prediction results?I also want to know how the "lines" are obtained in labe_new.json.Thank you for your enlightenment

**Comment by wvangansbeke** [comment]
Hi @jiaofang1222,

Yes. If you want to do lane detection in an end-to-end manner, you only the need the final lane line coefficients. I obtained those coefficients using the ground truth coordinates in the json files. After transforming the coordinates with a homography matrix to a bird's eye view perspective, I fit a second degree polynomial through these transformed points. I use those coefficients as ground truth for the regression. 

Hope this helps,
Wouter

**Comment by jiaofang1222** [comment]
But I use "np.polyfit" to fit the lane line parameters and run the model with my own dataset. I find that the training loss is very large and there will be problems such as the intersection of lane lines in the final lane line detection result. Instead, use your model directly. To test my data, although the prediction results are not very good, it is relatively better than the results obtained by training my own data. Do you have any better suggestions?

**Comment by wvangansbeke** [comment]
Hi @jiaofang1222, 

Hard to say what is wrong of course. Did you try a conventional segmentation approach? Maybe start getting that working. If that works, try augmenting it with the method in this repo. 

Kind regards.

**Comment by jiaofang1222** [comment]
Thank you for your time.I will try it!
